### PR TITLE
Configure `convertDeprecationsToExceptions="true"` in phpunit.xml.dist

### PR DESCRIPTION
**Pull request by TimWolla**

### CODE REVIEW

**Comment by TimWolla** [comment]
see also: php-http/client-integration-tests#57.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ jobs:
           php-version: '7.2'
           coverage: none
           extensions: mbstring, intl
+          ini-values: error_reporting=E_ALL
```
Comment (GrahamCampbell) [inline .github/workflows/ci.yml:21]: can we do this in bootstrap instead, so direct running locally has this, too?

Comment (GrahamCampbell) [reply .github/workflows/ci.yml:21]: eg https://github.com/laravel/framework/blob/v5.5.50/tests/bootstrap.php

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (TimWolla) [reply .github/workflows/ci.yml]: Good idea. It appears to work.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]
CI failing.

**Review by GrahamCampbell — APPROVED** [review]

**Pull request merged**